src/Draco.Compiler.Api/Diagnostics/Diagnostic.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Draco.Compiler.Api.Diagnostics;
+
+/// <summary>
+/// A diagnostic produced by the compiler.
+/// </summary>
+public sealed class Diagnostic
+{
+    /// <summary>
+    /// The location of the diagnostic.
+    /// </summary>
+    public Location Location { get; }
+
+    /// <summary>
+    /// The message explaining this diagnostics.
+    /// </summary>
+    public string Message =>
+        string.Format(this.internalDiagnostic.Format, this.internalDiagnostic.FormatArgs);
+
+    private readonly Internal.Diagnostics.Diagnostic internalDiagnostic;
+
+    internal Diagnostic(
+        Internal.Diagnostics.Diagnostic internalDiagnostic,
+        Location location)
+    {
+        this.internalDiagnostic = internalDiagnostic;
+        this.Location = location;
+    }
+
+    public override string ToString()
+    {
+        var sb = new StringBuilder();
```

Comment:
No need for a StringBuilder here. A regular interpolated string is more readable and use DefaultInterpolatedStringHandler which is more efficient than the StringBuilder.

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRjgAykCyAFAJYB2AF1I8ANKQjAAVjDBCAhgEpa9RqowB2UgBIARADEIEuVFo8AvmJpyzOgNwqGZwmaA==